state/block_cache.go

```diff
@@ -123,6 +126,44 @@ func (cache *BlockCache) SetStorage(addr Word256, key Word256, value Word256) {
 
 // BlockCache.storage
 //-------------------------------------
+// BlockCache.names
+
+func (cache *BlockCache) GetNameRegEntry(name string) *types.NameRegEntry {
+	entry, removed, _ := cache.names[name].unpack()
+	if removed {
+		return nil
+	} else if entry != nil {
+		return entry
+	} else {
+		entry = cache.backend.GetNameRegEntry(name)
+		cache.names[name] = nameInfo{entry, false, false}
+		return entry
+	}
+}
+
+func (cache *BlockCache) UpdateNameRegEntry(entry *types.NameRegEntry) {
+	name := entry.Name
+	// SANITY CHECK
+	_, removed, _ := cache.names[name].unpack()
+	if removed {
+		panic("UpdateNameRegEntry on a removed name")
```

Comment:
I think we need to make sure we catch these panics from the block_cache and tx_cache, I think in state/execution, otherwise a bad block proposal can maybe cause all network nodes to panic.  Is this true?  I will investigate further tonight.